src/box/lua/net_box.lua

```diff
@@ -749,21 +787,27 @@ function remote_methods:_request(method, opts, format, stream_id, ...)
     local res, err = transport:perform_request(timeout, buffer, skip_header,
                                                return_raw, on_push, on_push_ctx,
                                                format, stream_id, method, ...)
-    if err then
-        box.error(err)
-    end
     -- Try to wait until a schema is reloaded if needed.
     -- Regardless of reloading result, the main response is
     -- returned, since it does not depend on any schema things.
     if self.state == 'fetch_schema' then
         timeout = deadline and max(0, deadline - fiber_clock())
         self:wait_state('active', timeout)
     end
+    return res, err
+end
+
+function remote_methods:_request_xc(method, opts, format, stream_id, ...)
```

Comment:
I don't understand why you need to rename `_request` to `_request_xc`. What's wrong about using `pcall` in `ping`? It's worked fine so far so I don't see much point in changing it now. The downside of renaming `_request` to `_request_xc` is that it makes the patch more difficult to backport, because it affects a lot of unrelated files (tests).

Anyway, if we do want to add a method that works like `_request`, but doesn't throw an error, I think we'd better add `_request_impl` or `_request_no_xc`, without changing the existing code.

Reply:
> I don't understand why you need to rename _request to _request_xc

I was trying to follow the convention which we have in C++ code - when we want a function which both throws and not, we have `func` and `func_xc` - the latter throws.

> I think you'd better add _request_impl, without changing the existing code, and make _request call _request_impl after checking options 

I don't like here that it makes "impl" version not self-sufficient. Like it can't be called directly.

Reply:
> I was trying to follow the convention which we have in C++ code

I don't think we should adhere to our C++ practices here. The whole `func` and `func_xc` convention was supposed to be a temporary hack - until we get rid of exceptions completely. In Lua, in contrast to our C++ code, exceptions are allowed and often even preferred.

> I don't like here that it makes "impl" version not self-sufficient. Like it can't be called directly.

"impl" prefix suggests that the function's a part of the implementation and shouldn't be called directly. We could call it `_request_no_xc` though. It's just that I don't want to rename `_request` to `_request_xc` so as to keep the blast radius of this patch small.

Reply:
I agree with @locker. For me `impl` doesn't make function not-sufficient, it just says that we are not supposed to call it in general, it's preferable to use wrapper `_request`. I think we should stay with `_impl`. However, I don't insist on that option: there's not much differences (if we don't take into account the amount of affected code))